### PR TITLE
to support firmware & ramdisk as multiboot mods

### PR DESCRIPTION
**Pull request by mcao6**
1. refine multiboot related code, move to /boot.
2. firmware image files and ramdisk can be stitched in iasImage;
and they will loaded as multiboot modules.

Signed-off-by: Minggui Cao <minggui.cao@intel.com>

### CODE REVIEW

**Comment by dongyaozu** [comment]
For the initial patch, let us discuss in email to collect more comments?

**Comment by cmg-sky** [comment]
No problem, will do it later after an urgent task now.

File: Makefile [hunk 1]
```diff
@@ -88,6 +88,7 @@ D_SRCS += debug/sbuf.c
 C_SRCS += debug/printf.c
 D_SRCS += boot/acpi.c
```
Comment (anthonyzxu) [inline Makefile]: you probably need to do code rebase, it conflicts with https://github.com/projectacrn/acrn-hypervisor/commit/e8236e552d8aa07953cb728a78f1dff9903d9c9f

**Comment by mcao6** [comment]
Just rebase&updated, please merge it. Thanks!

**Comment by lijinxia** [comment]
verified +1

**Pull request merged**